src/main/java/com/ibm/eventstreams/kafkaconnect/plugins/xml/engines/StructToXmlBytes.java

```diff
@@ -233,19 +235,24 @@ private void processStruct(Document doc, Element parentElement, Struct source) {
 
                 case BYTES:
                     final Element bytesElement = doc.createElement(field.name());
-                    if (source.getBytes(field.name()) != null) {
-                        final byte[] bytes = source.getBytes(field.name());
-
-                        String strRepresentation;
-                        if (bytes.length == 1 && "xs:byte".equals(fieldSchema.doc())) {
-                            strRepresentation = Byte.toString(bytes[0]);
-                        }
-                        else {
-                            strRepresentation = new String(Base64.getEncoder().encode(bytes));
-                        }
-
-                        bytesElement.appendChild(doc.createTextNode(strRepresentation));
+                    final byte[] bytes ;
+                    String strRepresentation;
+
+                    if (Decimal.LOGICAL_NAME.equals(fieldSchema.name())){
+                       bytes = Decimal.fromLogical(fieldSchema, new BigDecimal(String.valueOf(source.get(field.name()))));
+                       strRepresentation = Decimal.toLogical(fieldSchema, bytes).toString();
+                    } else {
+                       bytes =  source.getBytes(field.name());
+                       if (bytes.length == 1 && "xs:byte".equals(fieldSchema.doc())) {
+                           strRepresentation = Byte.toString(bytes[0]);
+                       } else if (Decimal.LOGICAL_NAME.equals(fieldSchema.name())) {
```

Comment:
Unless I'm reading this incorrectly, I don't think this `if` check could ever pass 

Reading through this code, I can see we're inside the `else` branch of the `if (Decimal.LOGICAL_NAME.equals(fieldSchema.name()))` check from line 241 

That means this if check could never be satisfied, and that the line below is dead and unreachable code

Reply:
Sorry , missed to remove this block